### PR TITLE
HAAR-1344 sync user enabled flag end point

### PR DESCRIPTION
**Pull request by AlTurner-MOJ**

### CODE REVIEW

File: src/main/kotlin/uk/gov/justice/digital/hmpps/oauth2server/maintain/AuthUserSyncService.kt [hunk 1]
```diff
@@ -1,20 +1,39 @@
...
+import java.util.UUID
 
 @Service
 @Transactional
 class AuthUserSyncService(
   private val userRepository: UserRepository,
+  @Value("\${application.authentication.disable.login-days}") private val loginDaysTrigger: Int,
 ) {
   fun syncUserEmail(username: String, emailUpdate: UserEmailUpdate) {
     val user = userRepository.findByUsername(username).orElseThrow { UsernameNotFoundException("User with name $username not found") }
```
Comment (ravmoj) [inline src/main/kotlin/uk/gov/justice/digital/hmpps/oauth2server/maintain/AuthUserSyncService.kt:21]: all users are stored in upper case in DB...
may be change at repository level

Comment (ravmoj) [inline src/main/kotlin/uk/gov/justice/digital/hmpps/oauth2server/maintain/AuthUserSyncService.kt:21]: all users are stored in upper case in DB...
may be change at repository level

Comment (AlTurner-MOJ) [reply src/main/kotlin/uk/gov/justice/digital/hmpps/oauth2server/maintain/AuthUserSyncService.kt:21]: Discussed with @ravmoj and agreed this is related to other ticket and is being fixed under HAAR-1345

File: src/main/kotlin/uk/gov/justice/digital/hmpps/oauth2server/resource/api/AuthUserSyncController.kt [hunk 2]
```diff
@@ -81,11 +83,75 @@ class AuthUserSyncController(
...
+    @PathVariable
+    userId: UUID,
+    @Parameter(description = "The reason user made inactive.", required = true)
+    @RequestBody
+    userEnabledUpdate: UserEnabledUpdate,
+  ) {
+    authUserSyncService.syncUserEnabled(userId, userEnabledUpdate)
+
+    telemetryClient.trackEvent(
+      "AuthUserEnabledSync",
```
[the remark this answers is not in the document]
Comment (AlTurner-MOJ) [reply src/main/kotlin/uk/gov/justice/digital/hmpps/oauth2server/resource/api/AuthUserSyncController.kt:131]: Auth refers to them as Auth users, so stuck with that for consistency

**Review by ravmoj — APPROVED** [review]
lgtm

**Pull request merged**